knowledge_base/errors/database.rst

```diff
@@ -110,3 +110,197 @@ Once you figured out what's causing the issue, so it can be fixed later, you may
+                        ON blocking_locks.locktype = blocked_locks.locktype
+                        AND blocking_locks.DATABASE IS NOT DISTINCT FROM blocked_locks.DATABASE
+                        AND blocking_locks.relation IS NOT DISTINCT FROM blocked_locks.relation
+                        AND blocking_locks.page IS NOT DISTINCT FROM blocked_locks.page
+                        AND blocking_locks.tuple IS NOT DISTINCT FROM blocked_locks.tuple
+                        AND blocking_locks.virtualxid IS NOT DISTINCT FROM blocked_locks.virtualxid
+                        AND blocking_locks.transactionid IS NOT DISTINCT FROM blocked_locks.transactionid
+                        AND blocking_locks.classid IS NOT DISTINCT FROM blocked_locks.classid
+                        AND blocking_locks.objid IS NOT DISTINCT FROM blocked_locks.objid
+                        AND blocking_locks.objsubid IS NOT DISTINCT FROM blocked_locks.objsubid
+                        AND blocking_locks.pid != blocked_locks.pid
+                    JOIN pg_catalog.pg_stat_activity blocking_activity ON blocking_activity.pid = blocking_locks.pid
+                  WHERE NOT blocked_locks.GRANTED;
+
+           .. code-block:: SQL
+
+                SELECT a.datname,
+                         c.relname,
+                         l.transactionid,
+                         l.mode,
+                         l.GRANTED,
+                         a.usename,
+                         a.query,
+                         a.query_start,
+                         age(now(), a.query_start) AS "age",
+                         a.pid
+                    FROM  pg_stat_activity a
+                     JOIN pg_locks         l ON l.pid = a.pid
+                     JOIN pg_class         c ON c.oid = l.relation
+                    WHERE a.datname = current_database() AND a.pid <> pg_backend_pid()
+                    ORDER BY a.query_start;
+
+#. Find out where in Java the connections are used
+
+            These are the options available:
+
+            a) Create a thread dump (See :ref:`create-thread-dump`)
+            b) Terminate connections to get a stack trace in Java (See :ref:`force-close-db-connection`)
+            c) Enable leak detection:
+
+               .. code::
+
+                    oc set env -c nice dc/nice NICE2_HIKARI_leakDetectionThreshold=30000
+
+               This tells HikariCP to log connections that have been taken out of the pool for more than 30,000 ms.
+               HikariCP also logs a stack trace showing where the connection was obtained.
+
+               .. warning::
+
+                    Changing the leakDetectionThreshold automatically restarts Nice.
+
+
+Possible Measurements
+^^^^^^^^^^^^^^^^^^^^^
+
+a) Increase connection pool
+
+       .. parsed-literal::
+
+           oc set env -c nice dc/nice NICE2_HIKARI_maximumPoolSize=\ **15**
+
+       This increases the connection pool to **15** connections.
```

Comment:
a warning not to unnecessarily increase the pool size and a link to https://github.com/brettwooldridge/HikariCP/wiki/About-Pool-Sizing is probably useful.